### ISSUE TITLE
[Feature request]: Dreambooth support for text2video diffusion models

### ISSUE THREAD

**Issue by kabachuha** [huggingface/diffusers#2784]
### Model/Pipeline/Scheduler description

Now that [the first text2video pipeline is merged](https://github.com/huggingface/diffusers/pull/2738), the next frontier is to enable efficient fine-tuning of these models.

There's already Dreambooth method for the Diffusion-based text2img models, consisting of making a regularization dataset and then finetuning the model in a specific way on both the input and on the generated dataset.

Given that the nature of these models is essentially the same (as I know for maintaining the Auto1111 extension for this model), it can be enabled by shifting the Unet model from 2d to 3d and changing the class dataset pipeline from images to video. It would be really awesome to do 🙂

### Open source status

- [X] The model implementation is available
- [X] The model weights are available (Only relevant if addition is not a scheduler).

**Comment by patrickvonplaten** [comment]
This should actually be possible indeed! cc @sayakpaul here

**Comment by ExponentialML** [comment]
I've created an implementation  that should work if you want to pull from it. It's based off of the Tune-A-Video pipeline with some tweaks here and there.

https://github.com/ExponentialML/Text-To-Video-Finetuning

**Comment by sayakpaul** [comment]
This should indeed be possible. 

@kabachuha @ExponentialML would anyone of you like to open a PR? We're more than happy to guide throughout :)

**Comment by silvererudite** [comment]
I'd be very happy to take a jab at this if it's okay with everyone here, @sayakpaul can you please tell me what are the components I should put in the PR/the kind of PR that I should open for this task?

**Comment by sayakpaul** [comment]
Feel free to take a stab at it. 

The PR should be similar in terms of components to this ones: https://github.com/huggingface/diffusers/pull/2478

**Comment by sayakpaul** [comment]
@ExponentialML a friendly ping ^.

**Comment by adhikjoshi** [comment]
> @ExponentialML a friendly ping ^.

IMO too many text2video models coming. We should choose best one and provide updates.

**Comment by sayakpaul** [comment]
Having a training example would still be quite useful for the community, especially for use cases where personalization is required. 

> IMO too many text2video models coming.

That's even true for `diffusers`:

* https://github.com/huggingface/diffusers/pull/2455
* https://github.com/huggingface/diffusers/pull/2954

**Comment by ExponentialML** [comment]
> https://github.com/ExponentialML/Text-To-Video-Finetuning looks super cool!
> 
> BTW @ExponentialML would it be ok if we add a dreambooth for text-to-video fine-tuning script to diffusers' examples with you being the original author or would you prefer that we just link to your GitHub?

Sorry for the late response as my notifications are kind of messy :-). Sure, do as you wish.
